### DISCUSSION TITLE
Opening source code view at the start of the source code

### DISCUSSION THREAD

**Discussion by KarriKumaraLikeIt** [tinymce/tinymce#9863] [General]
Hello! Relative newcomer here.
When pressing the "source code" button in the TinyMCE editor, I'd like it to open at the beginning of the source code, not at the end.
Right now, if I open the source code for a long piece of text, the popup opens scrolled down to the very end of the text.
![image](https://github.com/user-attachments/assets/76b1909f-1ae6-419c-bf0e-855a6fa18931)

My guess would be that I would need to edit plugin.js for the "code" plugin. Any tips?

**Comment by TheSpyder** [comment]
There's nothing the plugin can do about this, it's using a plain textarea component and the implementation of that just uses `textarea.value = <data provided>`. There isn't an API to specify where the cursor should start in the data.
https://github.com/tinymce/tinymce/blob/124bb72f8d5ac9daa21a5b2bb9e866a47c06e6b9/modules/tinymce/src/plugins/code/main/ts/ui/Dialog.ts#L14-L17
